app/upgrades/v8/upgrades.go

```diff
@@ -42,26 +45,57 @@ func setOraclePairListEmpty(ctx sdk.Context, keepers *keepers.AppKeepers) {
 }
 
 func CreateUpgradeHandler(args upgrades.UpgradeHandlerArgs) upgradetypes.UpgradeHandler {
+	baseAppLegacySS := getLegacySubspaces(args.Keepers.ParamsKeeper)
+
 	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 		ctx.Logger().Info("running upgrade handler")
 
+		// ibc v4-to-v5
+		// -- nothing --
+
+		// ibc v5-to-v6
 		ctx.Logger().Info("migrating ics27 channel capability")
 		err := icaMigrations.MigrateICS27ChannelCapability(
 			ctx,
 			args.Codec,
-			args.CapabilityStoreKey,
-			args.CapabilityKeeper,
+			args.Keepers.GetKey(capabilitytypes.StoreKey),
+			args.Keepers.CapabilityKeeper,
 			lscosmostypes.ModuleName,
 		)
+		if err != nil {
+			return nil, err
+		}
 
+		// ibc v6-to-v7
+		// (optional) prune expired tendermint consensus states to save storage space
+		ctx.Logger().Info("pruning expired tendermint consensus states for ibc clients")
+		_, err = ibctmmigrations.PruneExpiredConsensusStates(ctx, args.Codec, args.Keepers.IBCKeeper.ClientKeeper)
+		if err != nil {
+			return nil, err
+		}
+
+		// sdk v45-to-v46
+		// -- nothing --
+
+		// sdk v46-to-v47
+		// Migrate Tendermint consensus parameters from x/params module to a dedicated x/consensus module.
+		ctx.Logger().Info("migrating tendermint consensus params")
+		baseapp.MigrateParams(ctx, baseAppLegacySS, args.Keepers.ConsensusParamsKeeper)
+
+		// Note: this migration is optional,
+		// You can include x/gov proposal migration documented in [UPGRADING.md](https://github.com/cosmos/cosmos-sdk/blob/main/UPGRADING.md)
+		// TODO(ajeet): do we need this optional migration?
+
+		ctx.Logger().Info("migrating lscsomos module")
+		err = args.Keepers.LSCosmosKeeper.Migrate(ctx)
```

Comment:
make sure you test this, afaik, it should be below migrations

Reply:
Migration is working.
But moved it after module migrations to be on safer side.